### PR TITLE
Add validation for password reset confirmation

### PR DESCRIPTION
**Pull request by bwndai**
The existing behavior of dj-rest-auth provided serializer PasswordResetConfirmSerializer validate the token and set the password in one go, this change extends the serializer and make that a two steps process.

### CODE REVIEW

File: src/planscape/planscape/settings.py [hunk 1]
```diff
@@ -108,6 +108,9 @@
 
 WSGI_APPLICATION = "planscape.wsgi.application"
 
+# TODO: This is for debugging email messages, and it needs to be removed.
+EMAIL_BACKEND='django.core.mail.backends.console.EmailBackend'
```
Comment (sterlingwellscaffeine) [inline src/planscape/planscape/settings.py:112]: According to the TODO message this needs to be removed or restructured into a settings.dev.py file.

File: src/planscape/planscape/urls.py [hunk 2]
```diff
@@ -29,4 +30,8 @@
     path('planscape-backend/dj-rest-auth/', include('dj_rest_auth.urls')),
     path('planscape-backend/dj-rest-auth/registration/',
          include('dj_rest_auth.registration.urls')),
+    # TODO: We don't actually need this since we'll be using
+    # a different endpoint.
```
Comment (sterlingwellscaffeine) [inline src/planscape/planscape/urls.py:34]: Is this comment still relevant?

File: src/planscape/users/forms.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+from allauth.account.forms import default_token_generator
+from allauth.account import app_settings
+from allauth.account.utils import user_pk_to_url_str
+
+from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+
+from dj_rest_auth.forms import AllAuthPasswordResetForm
+
+
```
Comment (sterlingwellscaffeine) [inline src/planscape/users/forms.py:11]: Please provide a class level comment describing the purpose of this class for future maintainers

File: src/planscape/users/serializers.py [hunk 4]
```diff
@@ -13,6 +15,27 @@ def custom_signup(self, request, user):
     user.last_name = self.validated_data.get('last_name', '')
     user.save(update_fields=['first_name', 'last_name'])
 
+class CustomPasswordResetSerializer(PasswordResetSerializer):
```
Comment (sterlingwellscaffeine) [inline src/planscape/users/serializers.py:18]: Please add comments for these classes to describing their purpose for future maintainers.

**Review by sterlingwellscaffeine — APPROVED** [review]

File: src/planscape/planscape/settings.py (hunk 1 above)
Comment (janicejl) [reply src/planscape/planscape/settings.py:112]: Rebasing to main should resolve this todo.
The TODO was resolved here: https://github.com/OurPlanscape/Planscape/pull/948

**Comment by runforyourlives** [comment]
With some of the desired changes lost in a bad git merge, we've added the diff to the original jira entry and will not merge this PR.

**Pull request closed without merging**